app/src/main/java/org/mozilla/fenix/onboarding/SyncCFRPresenter.kt

```diff
@@ -0,0 +1,74 @@
+package org.mozilla.fenix.onboarding
+
+import android.content.Context
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import mozilla.components.service.glean.private.NoExtras
+import org.mozilla.fenix.GleanMetrics.Onboarding
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.cfr.CFRPopup
+import org.mozilla.fenix.compose.cfr.CFRPopupProperties
+import org.mozilla.fenix.home.recentsyncedtabs.view.RecentSyncedTabViewHolder
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Delegate for handling sync onboarding CFR.
+ *
+ * @param context used for various Android interactions.
+ * @param settings used to read and write persistent user settings
+ * @param recyclerView will serve as anchor for the sync CFR
+ */
+
+class SyncCFRPresenter(
+    private val context: Context,
+    private val settings: Settings,
+    private val recyclerView: RecyclerView,
+) {
+
+    private var syncCFR: CFRPopup? = null
+
+    /**
+     * Check if [settings] allow sync CFR which may trigger showing one.
+     */
+    fun start() {
+        if (settings.showSyncCFR) {
```

Comment:
Maybe this check should be before we call start()?  This way we don't have to pass settings in and we don't have to create a `SyncCFRPresenter` if the user already saw the CFR.